app/components/Field/index.js

```diff
@@ -0,0 +1,126 @@
+import React, { createElement } from 'react';
+import PropTypes from 'prop-types';
+import isObject from 'utils/isObject';
+
+/**
+ * `Field` component.
+ *
+ * Used to connect any form field to the form data generated by the Form component.
+ *
+ * The prop `component` can be a Component, a stateless function component, or
+ * a string for DOM form fields (`input`, `select`). This is the component that will
+ * be rendered.
+ */
+class Field extends React.Component {
+  static propTypes = {
+    component: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
+    name: PropTypes.string.isRequired,
+    handleRef: PropTypes.func,
+  };
+
+  static defaultProps = {
+    component: 'input',
+    handleRef: null,
+  };
+
+  static contextTypes = {
```

Comment:
If you use contextTypes inside a component it becomes difficult to maintain. I highly suggest adding a HoC that brings this function as a property to Field.

Reply:
Sounds good. Fixed